## Hanna's Midterm
Data on more than 5,800 clergy who have been listed as credibly accused of sexual abuse in reports released by Catholic dioceses and religious orders from 2002-2020. 

### Import Python tools

In [34]:
import pandas as pd

### Read data

In [35]:
url = 'file:///Users/hannakang/Downloads/clergymembers.html'

In [36]:
df = pd.read_html(url)[0]

In [55]:
df.rename(columns=df.iloc[0], inplace = True)
df.drop([0], inplace = True)

### Basic information about data

In [56]:
df

id              name first_name  last_name  \
1     652690      Gerald Prinz     Gerald      Prinz   
2     652691     Gerard Kinane     Gerard     Kinane   
3     652692   Lawrence Cavell   Lawrence     Cavell   
4     652693    Patrick Kujawa    Patrick     Kujawa   
5     652694        Ramon Luce      Ramon       Luce   
...      ...               ...        ...        ...   
6749  652685  Claude Boudreaux     Claude  Boudreaux   
6750  652686        Dac Nguyen        Dac     Nguyen   
6751  652687       Dale Guidry       Dale     Guidry   
6752  652688      Daniel Poche     Daniel      Poche   
6753  652689   Etienne LeBlanc    Etienne    LeBlanc   

                                                 status birth_date ord_date  \
1     All priestly faculties permanently removed by ...        NaN     1968   
2     All priestly faculties permanently removed wit...        NaN     1973   
3     All priestly faculties permanently removed by ...        NaN     1989   
4                  Convicted and currently incarcerated        NaN     1996   
5                       Voluntarily left the priesthood        NaN     1995   
...                                                 ...        ...      ...   
6749                                           Deceased        NaN     1955   
6750                 Current status or location unknown        NaN     1988   
6751                                           Deceased        NaN     1982   
6752  All priestly faculties permanently removed by ...        NaN     1978   
6753  All priestly faculties permanently removed by ...        NaN     1971   

                                            assignments  \
1             Served at Saint Francis de Sales in Houma   
2                     Served at Saint Hilary in Mathews   
3                   Served at Holy Cross in Morgan City   
4     Served at Holy Cross in Morgan City and Saint ...   
5                     Served at Holy Savior in Lockport   
...                                                 ...   
6749                                                NaN   
6750        Served at Thanh Gia (Holy Family) in Amelia   
6751  Served at Holy Cross in Morgan City and Saint ...   
6752                Served at Holy Cross in Morgan City   
6753                      Served at Annunziata in Houma   

                     diocese_id                     diocese_name  \
1        Saint Gregory in Houma    and Saint Louis in Bayou Blue   
2                          2581       Diocese of Houma-Thibodaux   
3      Sacred Heart in Montegut  Saint Francis de Sales in Houma   
4                          2581       Diocese of Houma-Thibodaux   
5                          2581       Diocese of Houma-Thibodaux   
...                         ...                              ...   
6749                       2581       Diocese of Houma-Thibodaux   
6750                       2581       Diocese of Houma-Thibodaux   
6751                       2581       Diocese of Houma-Thibodaux   
6752    Sacred Heart in Cut Off        Saint Bernadette in Houma   
6753  Holy Cross in Morgan City             Holy Family in Dulac   

                                              list_name  \
1                                                  2581   
2     Disclosure of Names - The following priests ar...   
3                          Saint Genevieve in Thibodaux   
4     Disclosure of Names - The following priests ad...   
5     Disclosure of Names - The following priests ar...   
...                                                 ...   
6749  Disclosure of Names - The following priests ar...   
6750  Disclosure of Names - The following priests ar...   
6751  Disclosure of Names - The following priests ad...   
6752                    Saint Francis de Sales in Houma   
6753                              Saint Eloi in Theriot   

                                                   slug  \
1                            Diocese of Houma-Thibodaux   
2     ranging from inappropriate ph

### How many unique counts of sexual assault were reported?

In [57]:
len(df)

6753

### What is the distribution?

In [58]:
df.describe()

id           name first_name last_name    status birth_date  \
count     6753           6753       6752      6718      6109       1893   
unique    6753           5897        796      4383      1161        297   
top     651122  Robert Poandl       John    Murphy  Deceased       1946   
freq         1             10        576        27      2576         49   

       ord_date       assignments diocese_id                diocese_name  \
count      3506              3985       6686                        6741   
unique      146              2269       2765                        2826   
top        1955  Assistant Pastor       2496  Archdiocese of Los Angeles   
freq         97                99        301                         301   

                                                list_name  \
count                                                6753   
unique                                               2833   
top     CONSOLIDATED LIST OF PERSONS NAMED IN THE REPO...   
freq                                                  297   

                                           slug pp_id  
count                                      6751  6750  
unique                                     5038  5238  
top     WITH CURRENT STATUS THROUGH NOVEMBER 30  2018  
freq                                        297   297

##### = there were 6,754 unique counts of sexual assault, with Robert Poandl recording the highest count of sexual assault combined: 10. 2,576 of the accused clergymembers are deceased, the highest percentage of the current status of these accused clergymembers. Most of them held assistant pastor positions, and 301 of the assaults occurred at the Archdiocese of Los Angeles. 

### when did the clergy get ordinated? (oldest)

In [59]:
df.sort_values('ord_date', ascending=True).head()

id                       name first_name           last_name  \
4485   70998           Frederick George  Frederick              George   
4091   70466       Joseph Maxime Mooney     Joseph       Maxime Mooney   
5653   65865           Joseph T. Neeson     Joseph              Neeson   
2293  652166          Joseph Thibaudeau     Joseph          Thibaudeau   
5695   65935  Pierre Celestin Cambiaire     Pierre  Celestin Cambiaire   

                           status      birth_date ord_date  \
4485  Left religious order (1998)       Dismissed   (2004)   
4091              Deceased (1965)  Retired (1956)     1886   
5653                     Deceased             NaN     1890   
2293                     Deceased            1869     1896   
5695                     Deceased            1871     1898   

                assignments                   diocese_id  \
4485                    NaN                         1975   
4091                   1916                St. Johnsbury   
5653          December 1890    Associate Pastor/Chaplain   
2293             St. Jean's  Duluth - 6/30/1910 - ?/1943   
5695  Sacred Heart of Jesus          Cameron; St. George   

                               diocese_name               list_name  \
4485  St. Michael Catholic Church (Gastonia                   N.C.)   
4091                             Notre Dame             (Assistant)   
5653                       Associate Pastor  Saint Alphonsus Parish   
2293                                   2560       Diocese of Duluth   
5695                Baton Rouge; St. Joseph   Grosse-Tete; St. Leon   

                                                   slug  \
4485                                               2548   
4091                                1917-22; Burlington   
5653                                     Daviess County   
2293  Clergy with credible claims against them conce...   
5695                                          Leonville   

                                           pp_id  
4485                        Diocese of Charlotte  
4091                           St. Joseph Church  
5653                             KY and Chaplain  
2293  joseph-thibaudeau-diocese-of-duluth-652166  
5695                                        2501